bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanClassicDeviceHandler.java

```diff
@@ -98,12 +94,10 @@ public void thingUpdated(Thing thing) {
 
         // it seems that there does not exist a channel update callback
         // => remove all listeners and add them again
-        while (!channelConfigById.isEmpty()) {
-            removeListener(getThing().getChannel(channelConfigById.keys().nextElement()));
-        }
+        getBridgeHandler().removePacketListener(this);
 
-        getLinkedChannels().forEach(c -> {
-            if (!addListener(c)) {
+        this.getThing().getChannels().forEach(c -> {
+            if (isLinked(c.getUID().getId()) && !addListener(c)) {
```

Comment:
See above:

```suggestion
            if (isLinked(c.getUID()) && !addListener(c)) {
```

Reply:
Changed this and other occurrences of ChannelId => ChannelUID